hack/make/debug/debug.mk

```diff
@@ -0,0 +1,24 @@
+## Prepares all necessary CSI server resources for debugging (e.g. set limits, change build flags and startup command, ...)
+debug/prepare/csi-server:
+	git apply hack/make/debug/csi-server.patch
+
+## Prepares all necessary CSI provisioner resources for debugging (e.g. set limits, change build flags and startup command, ...)
+debug/prepare/csi-provisioner:
+	git apply hack/make/debug/csi-provisioner.patch
+
+## Forwards the CSI server port to localhost:40000
+debug/tunnel/csi-driver:
```

Comment:
maybe I am incorrect but wouldn't you need to rebuild/push the operator image? (you are not really mentioning that, but the patch messes with the dockerfile)

So IMO there should be a make command that does the "build part"
- and it would make the git patches safer in general (these patches are only needed for build and deploy)
   - in another comment I mention an alternative for the git patches, but this comment is still valid 😅  

wouldn't make targets like these make more sense:
- `make debug/build`: 
   1. apply patches to the Dockerfile
   2. build/push debug image
   3. remove patches from the Dockerfile
- `make debug/deploy`:
   1. apply patches to the yamls
   2. deploy the operator via helm
   3. remove patches from the yamls

Reply:
Updated the docu to make it more clear and also added a `make debug/build` command